### DISCUSSION TITLE
Unable to install .zip Library on Arduino IDE

### DISCUSSION THREAD

**Discussion by ShwetaKiranTotla** [pschatzmann/arduino-audio-tools#1513] [Q&A] [answered]
I want to receive audio via Bluetooth from phone and listen to the output from the internal DAC on headphones. To do the same, I am trying to add this Library to Arduino IDE, when I do the same, I get an error saying:

Error: 13 INTERNAL: Library install failed: extracting archive: Create link A required privilege is not held by the client.

Full error message:
```
Error: 13 INTERNAL: Library install failed: extracting archive: Create link  
C:\Users\Vitthalpyari\AppData\Local\Temp\2157286761\arduino-audio-tools-main\examples\examples-communication\a2dp\base-adc-a2dp: symlink ....\examples-basic-api\base-adc-a2dp  
C:\Users\Vitthalpyari\AppData\Local\Temp\2157286761\arduino-audio-tools-main\examples\examples-communication\a2dp\base-adc-a2dp: A required privilege is not held by the client.
```
Any help is appreciated.
Thanks!

**Comment by pschatzmann** [comment, marked as answer]
[In Windows, you need to have the permission to create symbolc links](https://blogs.windows.com/windowsdeveloper/2016/12/02/symlinks-windows-10/#oTepts0M8148xRoj.97) or [this](https://portal.perforce.com/s/article/3472#:~:text=Create%20Symbolic%20Links%20Privilege&text=Open%20Control%20Panel%2D%3EAdministrative%20Tools,Administrators%20group%2C%20add%20the%20user)

Alternatively download the zip and extract it yourself.

**Reply by ShwetaKiranTotla** [reply]
Sir,
Thank you for the reply. I have extracted the zip file in the Arduino Library folder in documents. After this, I tried to run the code for Audio sink that gives output to internal DAC and got the following error:
```
E:\1_test.ino:5:32: error: no matching function for call to 'BluetoothA2DPSink::BluetoothA2DPSink(audio_tools::AnalogAudioStream&)'
 BluetoothA2DPSink a2dp_sink(out);
                                ^
In file included from E:\1_test.ino:2:
c:\Documents\Arduino\libraries\ESP32-A2DP\src/BluetoothA2DPSink.h:95:5: note: candidate: 'BluetoothA2DPSink::BluetoothA2DPSink()'
     BluetoothA2DPSink();
     ^~~~~~~~~~~~~~~~~
c:\Documents\Arduino\libraries\ESP32-A2DP\src/BluetoothA2DPSink.h:95:5: note:   candidate expects 0 arguments, 1 provided
c:\Documents\Arduino\libraries\ESP32-A2DP\src/BluetoothA2DPSink.h:63:7: note: candidate: 'constexpr BluetoothA2DPSink::BluetoothA2DPSink(const BluetoothA2DPSink&)'
 class BluetoothA2DPSink : public BluetoothA2DPCommon {
       ^~~~~~~~~~~~~~~~~
c:\Documents\Arduino\libraries\ESP32-A2DP\src/BluetoothA2DPSink.h:63:7: note:   no known conversion for argument 1 from 'audio_tools::AnalogAudioStream' to 'const BluetoothA2DPSink&'

exit status 1

Compilation error: no matching function for call to 'BluetoothA2DPSink::BluetoothA2DPSink(audio_tools::AnalogAudioStream&)'
```
Could you please let me know how to fix this and where I might have gone wrong. Thank you!

**Comment by pschatzmann** [comment]
https://github.com/pschatzmann/arduino-audio-tools/wiki/Optional-Libraries
